ipywidgets/widgets/widget_output.py

```diff
@@ -68,3 +69,36 @@ def _flush(self):
         """Flush stdout and stderr buffers."""
         sys.stdout.flush()
         sys.stderr.flush()
+
+    def _append_stream_output(self, s, stream_name):
+        """Append a stream output."""
+        self.outputs += (
+            {'output_type': 'stream', 'name': stream_name, 'text': s},
+        )
+
+    def append_stdout(self, text):
+        """Append text to the stdout stream."""
+        self._append_stream_output(text, stream_name='stdout')
+
+    def append_stderr(self, text):
+        """Append text to the stderr stream."""
+        self._append_stream_output(text, stream_name='stderr')
+
+    def append_display_data(self, display_object):
+        """Append a display object as an output.
+
+        Parameters
+        ----------
+        display_object : IPython.core.display.DisplayObject
+            The object to display (e.g., an instance of
+            `IPython.display.Markdown` or `IPython.display.Image`).
+        """
+        fmt = InteractiveShell.instance().display_formatter.format
```

Comment:
This is based on [this line in IPython.core.display.py](https://github.com/ipython/ipython/blob/36ab971deab9c2822a1558caaa3ba99a49deb789/IPython/core/display.py#L291).  It seemed easier to follow how the `display` function works, where a user can pass, e.g., an `Image` object rather than asking users to provide the "raw" output format that this format function provides.  If there's a better way to get this formatting functionality, let me know.